internal/saucecloud/xcuitest.go

```diff
@@ -150,27 +159,36 @@ func (r *XcuitestRunner) calculateJobsCount(suites []xcuitest.Suite) int {
 }
 
 // archiveAppsToIpaIfRequired checks if apps are a .ipa package. Otherwise, it generates one.
-func archiveAppsToIpaIfRequired(appPath, testAppPath string) (archivedAppPath string, archivedTestAppPath string, archivedErr error) {
-	archivedAppPath = appPath
-	archivedTestAppPath = testAppPath
-	var err error
+func archiveAppsToIpaIfRequired(project *xcuitest.Project) (err error) {
+	appPath := project.Xcuitest.App
 	if !strings.HasSuffix(appPath, ".ipa") {
-		archivedAppPath, err = archiveAppToIpa(appPath)
+		project.Xcuitest.App, err = archiveAppToIpa(appPath)
 		if err != nil {
 			log.Error().Msgf("Unable to archive %s to ipa: %v", appPath, err)
-			archivedErr = fmt.Errorf("unable to archive %s", appPath)
+			err = fmt.Errorf("unable to archive %s", appPath)
 			return
 		}
 	}
-	if !strings.HasSuffix(testAppPath, ".ipa") {
-		archivedTestAppPath, err = archiveAppToIpa(testAppPath)
-		if err != nil {
-			log.Error().Msgf("Unable to archive %s to ipa: %v", testAppPath, err)
-			archivedErr = fmt.Errorf("unable to archive %s", testAppPath)
-			return
+	cache := map[string]string{}
+	for i, s := range project.Suites {
+		if !strings.HasSuffix(s.TestApp, ".ipa") {
+			if val, ok := cache[s.TestApp]; ok {
+				project.Suites[i].TestApp = val
+			} else {
+				var testAppPath string
+				testAppPath, err = archiveAppToIpa(s.TestApp)
+				if err != nil {
+					log.Error().Msgf("Unable to archive %s to ipa: %v", s.TestApp, err)
+					err = fmt.Errorf("unable to archive %s", s.TestApp)
+					return
+
+				}
+				project.Suites[i].TestApp = testAppPath
+				cache[s.TestApp] = testAppPath
+			}
```

Comment:
Removing 2 levels of nesting, as well as some slight error passing improvement.

```suggestion
		if strings.HasSuffix(s.TestApp, ".ipa") {
			continue
		}

		if val, ok := cache[s.TestApp]; ok {
			project.Suites[i].TestApp = val
			continue
		}
			
		var testAppPath string
		testAppPath, err = archiveAppToIpa(s.TestApp)
		if err != nil {
			log.Error().Msgf("Unable to archive %s to ipa: %v", s.TestApp, err)
			return fmt.Errorf("unable to archive %s: %w", s.TestApp, err)
		}
		project.Suites[i].TestApp = testAppPath
		cache[s.TestApp] = testAppPath
```